src/FSharp.Control.TaskSeq/TaskSeqBuilder.fs

```diff
@@ -657,6 +657,42 @@ module HighPriority =
                     sm.Data.current <- ValueNone
                     false)
 
+        member inline _.Bind
+            (
+                asyncSource: Async<'TResult1>,
+                continuation: ('TResult1 -> ResumableTSC<'T>)
+            ) : ResumableTSC<'T> =
+            ResumableTSC<'T>(fun sm ->
+                let mutable awaiter =
+                    Async
+                        .StartAsTask(asyncSource, cancellationToken = sm.Data.cancellationToken)
```

Comment:
why hop to threadpool ?

Reply:
What’s the (better) alternative @bartelink?

Reply:
Yep, it does the same: https://github.com/dotnet/fsharp/blob/d91b6c5c97accf363d135d1f99816410da4ec341/src/FSharp.Core/tasks.fs#L462

Reply:
I would think that `Async.StartImmediateAsTask` would be correct - you're already in a context where you're running as a `task` in a valid context so no need to queue to be re-dispatched from the threadpool, potentially losing one's timeslice? (IIRC the logic and wrapping is identical aside from the extra dispatch hop)
But hard to argue against doing the same as `task` does

Reply:
StartAsTask does a QueueAsync like Async.Start and Async.StartChild:
https://github.com/dotnet/fsharp/blob/main/src/FSharp.Core/async.fs#L1200
https://github.com/dotnet/fsharp/blob/main/src/FSharp.Core/async.fs#L1171

The only reason you'd prefer to queue rather than do it inline it is if you have a better thing to do on this thread right after you offload and/or you can. achieve better fairness by routing each piece of work through the thread pool each time?

Reply:
Seaching StartAsTask vs StarImmediateTask, this seems to be the only useful hit https://github.com/fsharp/fslang-suggestions/issues/521

Reply:
These are very acute and interesting points, I decided to record them separately (they may get lost in a merged PR): https://github.com/fsprojects/FSharp.Control.TaskSeq/issues/135.

Reply:
Thanks, that's great digging...